src/adagrams.js

```diff
@@ -1,15 +1,124 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = {...LETTER_POOL};
+  const TenStringsArray = []
+  for(let i = 0; i < 10; i++) {
+    const availableLetters = Object.keys(letterPool).filter(
+      (letter) => letterPool[letter] > 0
+  );
+    const randomIndex = Math.floor(Math.random()*availableLetters.length);
+    const chosenLetter = availableLetters[randomIndex];
+    TenStringsArray.push(chosenLetter);
+
+    letterPool[chosenLetter] -= 1;
+  }
+  return TenStringsArray;
```

Comment:
In this function right now you are not leveraging the respective weights of each letter, how could modify your logic to use them?